File: g2.cpp
from PIL import Image, ImageDraw
from easyinput import read, read_many

# Dibuixa un cercle amb el diametre i color donats, tal que la cantonada 
# esquerra superior del quadrat que el conté es troba al píxel (x,y)
def DibuixaCercle(x, y, diametre, color):
    dib.ellipse([(x, y), (x + diametre - 1, y + diametre - 1)], color)



# Mesures de la imatge
MARGE = 5
MARGE_ENTRE_BOLES = 10
DIAMETRE = 75
amplada = 3 * DIAMETRE + 2 * MARGE + 2 * MARGE_ENTRE_BOLES
alcada = DIAMETRE + 2 * MARGE

img = Image.new('RGB', (amplada, alcada), "AntiqueWhite")
dib = ImageDraw.Draw(img)

# Llegim la posició inicial 
v = [read(str) for _ in range(3)]

# Per cada instrucció, permutem els colors de les boles corresponents.
for instruccio in read_many(str):
    a, b = int(instruccio[0])-1, int(instruccio[-1])-1
    v[a], v[b] = v[b], v[a]

# Dibuixem
x_esquerra = MARGE
for color in v:
    DibuixaCercle(x_esquerra, MARGE, DIAMETRE, color)
    x_esquerra += DIAMETRE + MARGE_ENTRE_BOLES

img.save('output.png')

File: final/g2.py
from PIL import Image, ImageDraw
from easyinput import read

# Dibuixa un rectangle:
#    nw = cantonada superior esquerra
#    se = cantonada inferior dreta
def DibuixaRectangle(nw, se, color):
	dib.polygon([nw, (nw[0], se[1]), se, (se[0], nw[1])], color)

n = read(int)
color_f = [read(str) for _ in range(n)]
m = read(int)
color_s = [read(str) for _ in range(m)]

img = Image.new('RGB', (100*m, 150*n), 'Grey')
dib = ImageDraw.Draw(img)

for i in range(n):
	for j in range(m):
		x = 100*j
		y = 150*i
		dib.ellipse([(x + 35, y + 5), (x + 64, y + 34)], "AntiqueWhite")
		DibuixaRectangle((x + 5, y + 40), (x + 94, y + 54), color_s[j])
		DibuixaRectangle((x + 30, y + 55), (x + 69, y + 94), color_s[j])
		dib.polygon([(x + 30, y + 100), (x + 69, y + 100), (x + 79, y + 144), (x + 20, y + 144)], color_f[i])

img.save('output.png')
